Kuzzle.Tests/KuzzleTest.cs

```diff
@@ -0,0 +1,183 @@
+using Xunit;
+using Newtonsoft.Json.Linq;
+using KuzzleSdk.Exceptions;
+using Kuzzle.Tests.Protocol;
+using KuzzleSdk.Protocol;
+using KuzzleSdk.API;
+using System;
+using System.Threading.Tasks;
+using Moq;
+
+namespace Kuzzle.Tests {
+  public class KuzzleTest {
+    private readonly KuzzleSdk.Kuzzle _kuzzle;
+    private readonly ProtocolMock _protocol;
+
+    public KuzzleTest() {
+      _protocol = new ProtocolMock();
+      _kuzzle = new KuzzleSdk.Kuzzle(_protocol.MockedObject);
+    }
+
+    [Fact]
+    public void DispatchTokenExpiredTest() {
+      bool eventDispatched = false;
+
+      _kuzzle.TokenExpired += delegate() {
+        eventDispatched = true;
+      };
+
+      _kuzzle.DispatchTokenExpired();
+
+      Assert.Null(_kuzzle.AuthenticationToken);
+      Assert.True(eventDispatched);
+    }
+
+    [Fact]
+    public void KuzzleConstructorTest() {
+      Assert.NotNull(_kuzzle.Auth);
+      Assert.NotNull(_kuzzle.Collection);
+      Assert.NotNull(_kuzzle.Document);
+      Assert.NotNull(_kuzzle.Realtime);
+      Assert.NotNull(_kuzzle.Server);
+
+      Assert.NotNull(_kuzzle.Version);
+      Assert.NotNull(_kuzzle.InstanceId);
+    }
+
+    [Fact]
+    public async void ConnectAsyncTest() {
+      await _kuzzle.ConnectAsync();
+
+      _protocol.Mock.Verify(m => m.ConnectAsync(), Times.Once);
+    }
+
+    [Fact]
+    public void DisconnectTest() {
+      _kuzzle.Disconnect();
+
+      _protocol.Mock.Verify(m => m.Disconnect(), Times.Once);
+    }
+
+
+    [Fact]
+    public void QueryAsyncTest() {
```

Comment:
You should use a query generator, and test what happens if the passed query is null for instance.

FYI I just checked the code, and if a null query is provided to QueryAsync, the SDK crashes.

You should also include a query with a `volatile` object already set... and  even a `volatile` property that is not a JObject (I checked, and this will also make the SDK crash)